clmm/support/mock_data.py

```diff
@@ -332,12 +284,8 @@ def _generate_galaxy_catalog(cluster_m, cluster_z, cluster_c, cosmo, ngals,
     angsep, posangle = c_cl.separation(c_gal).rad, c_cl.position_angle(c_gal).rad
 
     posangle += 0.5*np.pi
-    if np.iterable(posangle):
-        posangle[posangle > np.pi] -= 2*np.pi
-        posangle[angsep == 0] = 0
-    else:
-        posangle -= 2*np.pi if posangle > np.pi else 0
-        posangle = 0 if angsep == 0 else posangle
+    posangle[posangle > np.pi] -= 2*np.pi
+    posangle[angsep == 0] = 0
```

Comment:
NOTE: This is the leftover code when we change from flat-sky to Astropy SkyCoord.
galaxy_catalog['ra'] and galaxy_catalog['dec'] are `astropy.table.column.Column` objects, so c_gal cannot return a scalar. `c_cl.isscalar` can be True when instantiated with both cluster_ra and cluster_dec being scalars. But after the operation `c_cl.position_angle` with non-scalar c_gal, `c_cl.isscalar` is False and therefore `np.iterable(posangle)` is always True.